File: java/src/test/java/org/kefir500/huifier/HuifierTest.java
package org.kefir500.huifier;

import org.junit.Test;

import static org.junit.Assert.*;

public class HuifierTest {

    @Test
    public void testRandomWords() {
        Huifier huifier = new Huifier();
        assertEquals("Хуилка", huifier.huify("Вилка"));
        assertEquals("Хуелик", huifier.huify("Толик"));
        assertEquals("Хуяблоко", huifier.huify("Яблоко"));
        assertEquals("Хуясный", huifier.huify("Красный"));
        assertEquals("Хуюассан", huifier.huify("Круассан"));
        assertEquals("Хую", huifier.huify("Хрю"));
    }

    @Test
    public void testVowelRules() {
        Huifier huifier = new Huifier();
        assertEquals("Хуярка", huifier.huify("Арка"));
        assertEquals("Хуедут", huifier.huify("Едут"));
        assertEquals("Хуёжик", huifier.huify("Ёжик"));
        assertEquals("Хуиглу", huifier.huify("Иглу"));
        assertEquals("Хуекна", huifier.huify("Окна"));
        assertEquals("Хуюрна", huifier.huify("Урна"));
        assertEquals("Хуик", huifier.huify("Стык"));
        assertEquals("Хуепос", huifier.huify("Эпос"));
        assertEquals("Хуюрта", huifier.huify("Юрта"));
        assertEquals("Хуящик", huifier.huify("Ящик"));
    }

    @Test
    public void testLetterCase() {
        Huifier huifier = new Huifier();
        assertEquals("Хуилка", huifier.huify("Вилка"));
        assertEquals("Хуилка", huifier.huify("вилка"));
        assertEquals("Хуилка", huifier.huify("вилка"));
        assertEquals("Хуилка", huifier.huify("вИлКа"));
    }

    @Test
    public void testNormalization() {
        Huifier huifier = new Huifier();
        assertEquals("Хуй", huifier.huify(""));
        assertEquals("Хуй", huifier.huify(" "));
        assertEquals("Хуй", huifier.huify("   "));
        assertEquals("Хуилка", huifier.huify("  Вилка  "));
        assertEquals("Хуй", huifier.huify("Й"));
        assertEquals("Хуя", huifier.huify("А"));
    }
}

File: rust/tests/huifier_test.rs
extern crate huifier;

#[test]
fn test_random_words() {
    assert_eq!(huifier::huify("Вилка"), "Хуилка");
    assert_eq!(huifier::huify("Толик"), "Хуелик");
    assert_eq!(huifier::huify("Яблоко"), "Хуяблоко");
    assert_eq!(huifier::huify("Красный"), "Хуясный");
    assert_eq!(huifier::huify("Круассан"), "Хуюассан");
    assert_eq!(huifier::huify("Хрю"), "Хую");
}

#[test]
fn test_vowel_rules() {
    assert_eq!(huifier::huify("Арка"), "Хуярка");
    assert_eq!(huifier::huify("Едут"), "Хуедут");
    assert_eq!(huifier::huify("Ёжик"), "Хуёжик");
    assert_eq!(huifier::huify("Иглу"), "Хуиглу");
    assert_eq!(huifier::huify("Окна"), "Хуекна");
    assert_eq!(huifier::huify("Урна"), "Хуюрна");
    assert_eq!(huifier::huify("Стык"), "Хуик");
    assert_eq!(huifier::huify("Эпос"), "Хуепос");
    assert_eq!(huifier::huify("Юрта"), "Хуюрта");
    assert_eq!(huifier::huify("Ящик"), "Хуящик");
}

#[test]
fn test_letter_case() {
    assert_eq!(huifier::huify("Вилка"), "Хуилка");
    assert_eq!(huifier::huify("вилка"), "Хуилка");
    assert_eq!(huifier::huify("ВИЛКА"), "Хуилка");
    assert_eq!(huifier::huify("вИлКа"), "Хуилка");
}

#[test]
fn test_normalization() {
    assert_eq!(huifier::huify(""), "Хуй");
    assert_eq!(huifier::huify(" "), "Хуй");
    assert_eq!(huifier::huify("   "), "Хуй");
    assert_eq!(huifier::huify("  Вилка  "), "Хуилка");
    assert_eq!(huifier::huify("Й"), "Хуй");
    assert_eq!(huifier::huify("А"), "Хуя");
}
